utils/utils.go

```diff
@@ -58,16 +57,18 @@ func CheckStakeWeightExceedsThreshold(accumulatedSignatureWeight *big.Int, total
 
 // BigToHashSafe ensures that a bignum value is able to fit into a 32 byte buffer before converting it to a common.Hash
 // Returns an error if overflow/truncation would occur by trying to perfom this operation.
+// TODO is this function still needed? It works for negative numbers now.
```

Comment:
Looking at `common.BigToHash`, I think we can remove this function. `big.Int` doesn't guarantee the internal bytes will be representable as a uint256 (though the getters will truncate to the requested type), so we may want to add redundant validity checks at the callsites.

Reply:
> (though the getters will truncate to the requested type), so we may want to add redundant validity checks at the callsites.

can you clarify this, didn't get it. We aren't using `BigToHashSafe` anywhere anyways, so I think we can remove if we don't imagine using it.

Reply:
hmm rereading this comment, I actually prefer leaving this utils function. Right now we don't use `common.BigTohash` or the `BigToHashSafe`, but want to leave for future use, since `common.BigToHash` will just truncate it to 32 bytes, I think it's better to check for error.